webpack/assets/javascripts/react_app/routes/FiltersForm/FiltersForm.js

```diff
@@ -0,0 +1,271 @@
+import React, { useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+import { useDispatch, useSelector } from 'react-redux';
+import {
+  Form,
+  FormGroup,
+  Checkbox,
+  Alert,
+  Popover,
+  ActionGroup,
+  Button,
+} from '@patternfly/react-core';
+import { HelpIcon } from '@patternfly/react-icons';
+import { translate as __ } from '../../common/I18n';
+import SearchBar from '../../components/SearchBar';
+import {
+  setAutocompleteSearchQuery,
+  updateDisability,
+  updateController,
+} from '../../components/AutoComplete/AutoCompleteActions';
+import './FilterForm.scss';
+import { SelectPermissions } from './SelectPermissions';
+import { SelectResourceType } from './SelectResourceType';
+import { SelectRole } from './SelectRole';
+import { EMPTY_RESOURCE_TYPE, SEARCH_ID } from './FiltersFormConstants';
+import { Taxonomies } from './Taxonomies';
+import { APIActions } from '../../redux/API';
+import { selectAutocompleteSearchQuery } from '../../components/AutoComplete/AutoCompleteSelectors';
+import { addToast } from '../../components/ToastsList';
+
+export const FiltersForm = ({ roleName, roleId, isNew, data, history }) => {
+  const [role, setRole] = useState(roleId);
+  const [type, setType] = useState(EMPTY_RESOURCE_TYPE);
+  const [chosenPermissions, setChosenPermissions] = useState([]);
+  const [isUnlimited, setIsUnlimited] = useState(!!data['unlimited?']);
+  const [isOverride, setIsOverride] = useState(!!data['override?']);
+  const [isGranular, setIsGranular] = useState(false);
+  const [chosenOrgs, setChosenOrgs] = useState(
+    data.organizations?.map(o => o.id) || []
+  );
+  const [chosenLocations, setChosenLocations] = useState(
+    data.locations?.map(l => l.id) || []
+  );
+  const {
+    show_organizations: showOrgs = false,
+    show_locations: showLocations = false,
+  } = type;
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    if (data.search)
+      dispatch(setAutocompleteSearchQuery(data.search, SEARCH_ID));
+    // workaround a bug - initial search query is ignored
+  }, [data.search, dispatch]);
+
+  const autocompleteQuery = useSelector(state =>
+    selectAutocompleteSearchQuery(state, SEARCH_ID)
+  );
+  const submit = async () => {
+    const params = {
+      filter: {
+        role_id: role,
+        search: autocompleteQuery,
+        unlimited: isUnlimited,
```

Comment:
I have a role that had the unlimited checkbox selected.
when I try to unselect the checkbox and submit the form, it still loads as selected on the next time that I try to edit the role.

Reply:
Thanks, added a useeffect which will be removed with the new search pr

Reply:
not sure if that's an issue or made by our intention,
when search input has some value and you click on the `unlimited` checkbox, it turns disabled, but after the form submits, you'll see that the Role still has search and Unlimited = false.


https://user-images.githubusercontent.com/26363699/191269353-68a63acb-e8c5-4107-babf-db83e09632b9.mp4